pkg/daemon/client_test.go

```
@@ -168,3 +169,157 @@ func TestClient_Heartbeat_DecodeError(t *testing.T) {
+	}
+}
+
+func TestTelemetryDueError(t *testing.T) {
+	_, priv, err := crypto.GenerateSigningKeyPair()
+	if err != nil {
+		t.Fatalf("failed to generate signing key: %v", err)
+	}
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("internal error"))
```

golangci-lint (2.12.2)

[error] Error return value of `w.Write` is not checked

(errcheck)